### PR TITLE
IGAPP-593: Remove permission snackbar and fix iOS PN permission handling

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/routes/settings/components/SettingItem.js [hunk 1]
```diff
@@ -1,24 +1,29 @@
 // @flow
```
Comment (klinzo) [inline native/src/routes/settings/components/SettingItem.js:1]: Refactoring only, moved some functionality from `Settings` here: badges and switches

File: native/src/routes/settings/components/Settings.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
 // @flow
```
Comment (klinzo) [inline native/src/routes/settings/components/Settings.js:1]: Refactoring only, moved some functionality to `SettingsItems` here: badges and switches

File: native/src/routes/settings/__tests__/createSettingsSections.spec.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
+// @flow
```
Comment (SteffiStuffel) [inline native/src/routes/settings/__tests__/createSettingsSections.spec.js:1]: nice! Go for the OKR :rocket:

**Review by SteffiStuffel — APPROVED** [review]
Did regression test for settings page.

File: native/src/routes/settings/components/Settings.js [hunk 4]
```diff
@@ -56,135 +50,97 @@ const SectionHeader = styled.Text`
...
 
-    this.unsubscribeNavigationListener = this.props.navigation.addListener('focus', () => {
-      this.loadSettings()
+  useEffect(() => {
+    // Reload settings if navigating back from another route
+    return navigation.addListener('focus', () => {
+      loadSettings()
     })
-  }
+  }, [navigation])
```
Comment (Taggotty) [inline native/src/routes/settings/components/Settings.js:67]: No need to remove the listener on unmount?
Any reason why the [useFocusEffect](https://reactnavigation.org/docs/function-after-focusing-screen/#triggering-an-action-with-the-usefocuseffect-hook) hook doesn't work here?

Comment (klinzo) [reply native/src/routes/settings/components/Settings.js:67]: It is removed on unmount, `navigation.addListener` directly returns the function to remove the listener as mentioned here: https://reactnavigation.org/docs/navigation-events/#navigationaddlistener
`useFocusEffect` seems fine, I just didn't look into refactoring this even more and also did not know about this one. Will try it :+1:

Comment (klinzo) [reply native/src/routes/settings/components/Settings.js]: Works nicely, thanks!

**Pull request merged**